Full Site Editing: Add an hover transition to the Template block overlay

Copons opened the pull request:
#### Changes proposed in this Pull Request

* Add an hover transition to the Template block overlay.
  * The overlay appears in 0.2 seconds.
  * The button and the blur effect in 0.2 seconds, delayed by 0.7 seconds (as in: 0.5 seconds after the overlay appeared).
* Selecting the block makes the overlay "stick": it doesn't disappear on mouse out.
* Clicking on the "Edit" button makes the overlay "stick" too while the template editor loads.

| Before | After |
| - | - |
| ![Aug-22-2019 16-05-27](https://user-images.githubusercontent.com/2070010/63526634-9c508680-c4f7-11e9-9025-28a640c4c942.gif) | ![Aug-22-2019 18-24-37](https://user-images.githubusercontent.com/2070010/63535851-5650ee00-c50a-11e9-9c5b-4a198bc38fe0.gif) |

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Edit a page on an FSE site.
* Poke around a template part to see if everything works as expected, and the effect feels good.

Fixes #35437

apeatling commented:
@Copons The transitions and hover work perfectly, but I'm still seeing the grey background on grey button?

<img width="548" alt="Screen Shot 2019-08-22 at 9 18 10 AM" src="https://user-images.githubusercontent.com/1464705/63531670-e0bf3080-c4bd-11e9-9250-83810d7e0c94.png">

Copons commented:
> I'm still seeing the grey background on grey button

Oh shoot, I somehow didn't notice that @shaunandrews demo was basically a fade to white, instead to gray!
With the white-faded overlay, I should also make the borders visible again then.
Back to WIP!

Copons commented:
@apeatling I've updated the overlay and added a subtle border.
I've also realized that the gifs there look all white anyway, so here's an actual screen of what it looks like now:

![Screenshot 2019-08-22 at 18 28 13](https://user-images.githubusercontent.com/2070010/63536007-a29c2e00-c50a-11e9-93cd-bba8ff61cd6c.png)

apeatling commented:
Look great!
